From patchwork Wed Apr 24 21:22:18 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Nadav Amit <nadav.amit@gmail.com>
X-Patchwork-Id: 10915751
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 3931613B5
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 24 Apr 2019 21:22:45 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 259D628A02
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 24 Apr 2019 21:22:45 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 1792028C00; Wed, 24 Apr 2019 21:22:45 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id B583728A02
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 24 Apr 2019 21:22:44 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1730633AbfDXVWn (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 24 Apr 2019 17:22:43 -0400
Received: from mail-pg1-f194.google.com ([209.85.215.194]:35831 "EHLO
        mail-pg1-f194.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1725978AbfDXVWn (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 24 Apr 2019 17:22:43 -0400
Received: by mail-pg1-f194.google.com with SMTP id j24so346666pgk.2
        for <kvm@vger.kernel.org>; Wed, 24 Apr 2019 14:22:43 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=vUctaQuoQkYtMZ2FhoZfOBJG43z1/7K1NX2PWLjERQA=;
        b=q6hl17YZcvH9k2kS3QV0fQC1JTX9uH5EBDChXbCE3ReGEOMqlZ/8zA4YihFMAG6MR1
         M64z/qX+kDa0gueMPpKFO4OQAlPUYCOMsgZggDpTZ+liNNnl3QAhv2YTXUZo8PIL/dro
         5WBkHsTV2hLjlqvpelj54eyNStoGCnlrivyyMwnyniRYkN07M5u+xIa4uM5Vl3sD54i1
         7G782hRtOe7ld/gVpVTeoeT6VDbYCHRNafPIS+jD+PhG0JRtLWuqliuY6N4L3oKcxaRu
         sUPFLlsnZHj+BoBUrv/s75vld2MJvU9RRIAyD/P8R1Sl80a4GYpzl7CuoUc5mvUX8uTP
         EG9g==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=vUctaQuoQkYtMZ2FhoZfOBJG43z1/7K1NX2PWLjERQA=;
        b=AEDqEr1TYX1vl0ubc+vC6a5Mu7Zgi9maImvbLg47bTmK9ry3f92GCIBVmQStNT7XH3
         4TKvKaWD8ut/+In/4EzIKkVrQZcPdivdY92VH+6/zbvO1ofxK4YZ3b1KAmdCOGJaTLxn
         ENTDBL4JnuIrEgsxqj1Fvg1B/KcI2h0x68nCIz4NGs6Jdlvr71CfrySmmTirlLloL20G
         v0z8AbqF5RtX2Re+mj68t7ECXF+JCMilAKOUl3tmUuoxQ5YRaVSL+1ohXo+pGN6risDK
         muBSBwYI6v479OEyd/Eth0h/oKq+uzX0YcYs1Z+w9n+6+GNntnkhmEhtkdYHX7NJrJWO
         UnRw==
X-Gm-Message-State: APjAAAWAP1JO7ZC+9JSDbcJH661BzFWfy+LVhYw8EXhLG4XPA0jCX9M1
        gDYyxWfrBNWzndc5TlUtBk4=
X-Google-Smtp-Source: 
 APXvYqwxbbwojQwwSsELiQllQ4OwAqcOLYBNtEokF8Xw5NMCSFQ/luuInx+OD8cRMLzOMenbm94pyw==
X-Received: by 2002:a63:b305:: with SMTP id i5mr32838367pgf.274.1556140962248;
        Wed, 24 Apr 2019 14:22:42 -0700 (PDT)
Received: from htb-2n-eng-dhcp405.eng.vmware.com ([66.170.99.1])
        by smtp.gmail.com with ESMTPSA id
 h20sm8974085pgb.25.2019.04.24.14.22.41
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Wed, 24 Apr 2019 14:22:41 -0700 (PDT)
From: nadav.amit@gmail.com
To: Paolo Bonzini <pbonzini@redhat.com>
Cc: kvm@vger.kernel.org, Nadav Amit <nadav.amit@gmail.com>,
        Sean Christopherson <sean.j.christopherson@intel.com>
Subject: [kvm-unit-tests PATCH] x86: Allow xapic ID writes to silently fail
Date: Wed, 24 Apr 2019 14:22:18 -0700
Message-Id: <20190424212218.15230-1-nadav.amit@gmail.com>
X-Mailer: git-send-email 2.19.1
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

From: Nadav Amit <nadav.amit@gmail.com>

According to Intel SDM: "Some processors permit software to modify the
APIC ID.  However, the ability of software to modify the APIC ID is
processor model specific."

Allow this behavior not to cause failures.

Cc: Sean Christopherson <sean.j.christopherson@intel.com>
Signed-off-by: Nadav Amit <nadav.amit@gmail.com>
---
 x86/apic.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/apic.c b/x86/apic.c
index d1ed5ea..6772f3f 100644
--- a/x86/apic.c
+++ b/x86/apic.c
@@ -210,7 +210,7 @@ static void __test_apic_id(void * unused)
     newid = (id + 1) << 24;
     report("writeable xapic id",
             !test_for_exception(GP_VECTOR, do_write_apic_id, &newid) &&
-            id + 1 == apic_id());
+	    (id == apic_id() || id + 1 == apic_id()));
 
     if (!enable_x2apic())
         goto out;
